WordPress/gutenberg
# [RNMobile] Improve AztecWrapper RN talkback support

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/918
Aztec Android PR https://github.com/wordpress-mobile/AztecEditor-Android/pull/927
Gutenberg Mobile PR https://github.com/wordpress-mobile/gutenberg-mobile/pull/2630

## Description
Currently the RichText will focus automatically without needing double tapping. This was taking place due to the `AccessibilityDelegate` in `Aztec` requesting focus on hover, so that behavior was removed. This PR, resolves this by utilizing the mentioned fix within Aztec along with enabling the `RichText` component via the `ReactAztecManager` so that the `double-tap to activate` behavior becomes active.

## Testing
This was first tested in Aztec by ensuring that the `AztecText` view no longer auto focuses on hover. The commit hash of the Aztec PR was then utilized within Gutenberg to verify that the editor would no longer perform this auto-focusing.

For further testing here are the instructions. 
1. Enable TalkBack. 
2. Hover over a text component that's powered by the `RichText` such as a paragraph block. 
3. Notice that it only reads the contents of the block but it no longer shows a blinking cursor or opens the keyboard. 
4. Perform the announced action of double tapping to activate and then the cursor and the keyboard becomes active.

## Screenshots 
<kbd><img src="https://user-images.githubusercontent.com/1509205/93387592-194f4c00-f82f-11ea-8b6d-40995c5e82d3.gif" width="320"></kbd>

[Video Example](https://drive.google.com/file/d/1SPli0o0o9IpEyXO1V7pdl4vO8T9zdKj4/view?usp=sharing) 

## Concerns
The only change made in this PR I was wondering about is the removal of `setFocusable(false)` I tried to utilize git blame to figure out why this was added in the first place but I wasn't able to find [a commit with enough details](https://github.com/wordpress-mobile/react-native-aztec/commit/357a9a4312d4c842ea1b68f79fb56aff4773f399), hence I am pointing it out here. 
**Update** this was being discussed [here](https://github.com/wordpress-mobile/gutenberg-mobile/issues/14) as well. I will look into it more.

## Checklist:

## Review
* Currently this PR references an Aztec commit hash so this needs to be updated before merging.
